server/instance.go

```diff
@@ -13,9 +13,27 @@ import (
 	"github.com/labstack/echo/v4"
 )
 
+// dbaFlowMiddleware is a feature guard for bb.feature.dba-workflow
+func dbaFlowMiddleware(server *Server, next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		role := c.Get(getRoleContextKey()).(api.Role)
+		if role == api.Developer && server.feature(api.FeatureDBAWorkflow) {
+			return echo.NewHTTPError(http.StatusForbidden, "Developer has not access to instance")
```

Comment:
OK to forbid instance mutation operation, however GET operation is probably needed since Bytebase needs to fetch instance data on behalf of the end user to perform the UI.

For dba workflow, it's mostly the UI trick to hide the instance concept from developer.